src/core.lisp

```diff
@@ -1,3 +1,90 @@
 ;;;; Rasterization renderer.
 
 (in-package #:cl-raster/core)
+
+(defstruct depth-point
+  point
+  depth)
+
+(defun d-p-x (point)
+  (v:vx (depth-point-point point)))
+
+(defun d-p-y (point)
+  (v:vy (depth-point-point point)))
+
+(defun project-point-to-camera (camera point)
+  (let* ((O->p (v:v- (scene:camera-center camera) point))
+         (O->p* (v:v/ O->p
+                      (v:v. O->p
+                            (scene:camera-direction camera))))
+         (o*->p* (v:v- O->p* (scene:camera-direction camera)))
+         (x (v:v. (scene:camera-x-vector camera) o*->p*))
+         (y (v:v. (scene:camera-y-vector camera) o*->p*)))
+    (make-depth-point :point
+                      (v:vec2 (round (* (/ (1+ x) 2) (scene:camera-width camera)))
+                              (round (* (/ (1+ y) 2) (scene:camera-height camera))))
+                      :depth
+                      (v:v. O->p
+                            (v:vunit (scene:camera-direction camera))))))
+
+(defun trg-to-camera (camera trg)
```

Comment:
Maybe `triangle` instead of `trg`? And maybe with `project-` prefix similarly to `project-point-to-camera`?